src/scheduler/pbs_sched_bare.cpp

```diff
@@ -0,0 +1,294 @@
+ * @param[in]	sinfo	-	pbs universe we're going to loop over
+ * @param[out]	rerr	-	error bits from the last job considered
+ *
+ *	@return return code of last job scheduled
+ *	@retval -1	: on error
+ */
+int
+main_sched_loop_bare(int sd, server_info *sinfo)
+{
+	node_info **nodes = sinfo->nodes;
+	resource_resv **jobs = sinfo->jobs;
+	int ij;
+	int in = 0;
+
+	/* Algorithm:
+     * - Loop over all jobs, assume that they need just 1 ncpu to run, and
+     *  choose the next free node for it
+     */
+	for (ij = 0; jobs[ij] != NULL; ij++) {
+		char execvnode[PBS_MAXHOSTNAME + 1];
+		execvnode[0] = '\0';
+
+		/* Find the first free node and fill it */
+		for (; nodes[in] != NULL; in++) {
+			node_info *node = nodes[in];
+			schd_resource *ncpures = NULL;
+
+			if (node->is_busy || node->is_job_busy)
+				continue;
+
+			ncpures = find_resource(node->res, getallres(RES_NCPUS));
+			if (ncpures == NULL)
+				continue;
+
+			/* Assign a cpu on this node */
+			ncpures->assigned += 1;
+			if (dynamic_avail(ncpures) == 0) {
+				node->is_busy = 1;
+				node->is_job_busy = 1;
+				node->is_free = 0;
+			}
+
+			/* Create the exec_node for the job */
+			snprintf(execvnode, sizeof(execvnode), "(%s:ncpus=1)", node->name);
+
+			/* Send the run request */
+			send_run_job(sd, 0, jobs[ij]->name, execvnode);
+
+			break;
+		}
+		if (execvnode[0] == '\0') {
+			log_event(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, "",
+				  "No free nodes available, won't consider any more jobs");
+			break;
+		}
+	}
+	return SUCCESS;
+}
+
+/**
+ * @brief
+ *		scheduling_cycle - the controling function of the scheduling cycle
+ *
+ * @param[in]	sd	-	primary socket descriptor to the server pool
+ *
+ * @return	int
+ * @retval	0	: success/normal return
+ * @retval	-1	: failure
+ *
+ */
+
+static int
+scheduling_cycle_bare(int sd, const sched_cmd *cmd)
+{
+	server_info *sinfo;	    /* ptr to the server/queue/job/node info */
+	int rc = SUCCESS;	    /* return code from main_sched_loop() */
+
+	log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_REQUEST, LOG_DEBUG,
+		  "", "Starting Scheduling Cycle");
+
+	/* Decide whether we need to send "can't run" type updates this cycle */
+	if (time(NULL) - last_attr_updates >= sc_attrs.attr_update_period)
+		send_job_attr_updates = 1;
+	else
+		send_job_attr_updates = 0;
+
+	update_cycle_status(&cstat, 0);
+
+	/* create the server / queue / job / node structures */
+	if ((sinfo = query_server(&cstat, sd)) == NULL) {
+		log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_NOTICE,
+			  "", "Problem with creating server data structure");
+		end_cycle_tasks(sinfo);
+		return 0;
+	}
+
+	rc = main_sched_loop_bare(sd, sinfo);
+
+	end_cycle_tasks(sinfo);
+
+	return rc;
+}
+
+/**
+ * @brief
+ *		intermediate_schedule - responsible for starting/restarting scheduling
+ *		cycle.
+ *
+ * @param[in]	sd	-	primary socket descriptor to the server pool
+ *
+ * returns 0
+ *
+ */
+static int
+intermediate_schedule_bare(int sd, const sched_cmd *cmd)
+{
```

Comment:
You probably don't need intermediate_schedule_bare().  It's to handle rare cases where we want to restart the scheduling cycle.